### PR TITLE
Fix host_in_group placeholder

### PR DESCRIPTION
**Pull request by PhilippSiegmantel**

### CODE REVIEW

File: src/ui/components.tsx [hunk 1]
```diff
@@ -176,10 +176,10 @@ export const CheckMkSelectNegatable = <T extends RequestSpecNegatableOptionKeys>
           label={label}
           autocompleter={autocompleter}
           onChange={onValueChange}
-          value={value.value}
+          value={value.value === '' ? undefined : value.value}
         />
       </InlineField>
-      <Checkbox label="Negate" value={value !== undefined ? value.negated : false} onChange={onNegateChange} />
+      <Checkbox label="Negate" value={value.negated} onChange={onNegateChange} />
```
Comment (BenediktSeidl) [inline src/ui/components.tsx:182]: can you add a comment to the commit message mentioning this change? this seems to be unrelated to the current commit message?

File: src/ui/components.tsx [hunk 2]
```diff
@@ -176,10 +176,10 @@ export const CheckMkSelectNegatable = <T extends RequestSpecNegatableOptionKeys>
           label={label}
           autocompleter={autocompleter}
           onChange={onValueChange}
-          value={value.value}
+          value={value.value === '' ? undefined : value.value}
```
Comment (BenediktSeidl) [inline src/ui/components.tsx:179]: ah! sorry, to mention this so late, but could there be another/better place to fix this?
why is the value an empty string and not undefined in the first place? could this be fixable with a better default value andtyping?

Comment (BenediktSeidl) [reply src/ui/components.tsx:179]: can you give it a try? if it's too much work, or looks too ugly, we can still go back to this option.

Comment (PhilippSiegmantel) [reply src/ui/components.tsx]: Thank you for pushing me, it changed exaclty nothing.

File: src/ui/components.tsx [hunk 3]
```diff
@@ -30,7 +30,7 @@ interface CheckMkAsyncSelectProps<T> {
   width?: number;
   autocompleter: (prefix: string) => Promise<Array<SelectableValue<NonNullable<T>>>>;
   onChange: (value: T) => void;
-  value: T;
+  value?: T;
```
Comment (BenediktSeidl) [inline src/ui/components.tsx:33]: sorry, that i find something new on each iteration...

this seems to be a bit broad to me. what do you think of the following change:

```diff
diff --git a/src/ui/components.tsx b/src/ui/components.tsx
index 1615d15..4a42954 100644
--- a/src/ui/components.tsx
+++ b/src/ui/components.tsx
@@ -30,7 +30,7 @@ interface CheckMkAsyncSelectProps<T> {
   width?: number;
   autocompleter: (prefix: string) => Promise<Array<SelectableValue<NonNullable<T>>>>;
   onChange: (value: T) => void;
-  value?: T;
+  value: T;
   inputId: string; // make the InlineField magic do its work // TODO: find better solution
 }
 
@@ -155,7 +155,7 @@ export const CheckMkSelectNegatable = <T extends RequestSpecNegatableOptionKeys>
         }
       : { ...props.value };
 
-  const onValueChange = (newValue: string) => {
+  const onValueChange = (newValue: string | undefined) => {
     if (newValue === undefined) {
       throw 'this is an internal error, please report';
     }
@@ -171,7 +171,7 @@ export const CheckMkSelectNegatable = <T extends RequestSpecNegatableOptionKeys>
   return (
     <HorizontalGroup>
       <InlineField label={label} labelWidth={14}>
-        <CheckMkAsyncSelect
+        <CheckMkAsyncSelect<string | undefined>
           inputId={`input_${props.label}`}
           label={label}
           autocompleter={autocompleter}
```

this would be a more specific to the thing we want to change?!

Comment (PhilippSiegmantel) [reply src/ui/components.tsx:33]: I like it! A lot!

**Review by BenediktSeidl — APPROVED** [review]
